cmd/auth/login.go

```diff
@@ -39,35 +58,30 @@ func newLoginCommand(persistentAuth *auth.PersistentAuth) *cobra.Command {
 
 	var loginTimeout time.Duration
 	var configureCluster bool
+	var profileName string
 	cmd.Flags().DurationVar(&loginTimeout, "timeout", auth.DefaultTimeout,
 		"Timeout for completing login challenge in the browser")
 	cmd.Flags().BoolVar(&configureCluster, "configure-cluster", false,
 		"Prompts to configure cluster")
+	cmd.Flags().StringVarP(&profileName, "profile", "p", "", `Name of the profile.`)
 
 	cmd.RunE = func(cmd *cobra.Command, args []string) error {
 		ctx := cmd.Context()
 
-		var profileName string
-		profileFlag := cmd.Flag("profile")
-		if profileFlag != nil && profileFlag.Value.String() != "" {
-			profileName = profileFlag.Value.String()
-		} else if cmdio.IsInTTY(ctx) {
-			prompt := cmdio.Prompt(ctx)
-			prompt.Label = "Databricks Profile Name"
-			prompt.Default = persistentAuth.ProfileName()
-			prompt.AllowEdit = true
-			profile, err := prompt.Run()
+		// If the user has not specified a profile name, prompt for one.
+		if profileName == "" {
+			var err error
+			profileName, err = promptForProfile(ctx, persistentAuth.DefaultProfileName())
 			if err != nil {
 				return err
 			}
-			profileName = profile
 		}
 
+		// Set the host based on the provided arguments and flags.
 		err := setHost(ctx, profileName, persistentAuth, args)
 		if err != nil {
 			return err
 		}
-		defer persistentAuth.Close()
```

Comment:
Why move this?

Reply:
Set host only modifies the `persistentAuth.Host` value. Ifigured we don't need to close the network listener before `persistentAuth.Challenge()` is called.

To be honest I did not take the time to really understand what's happening the challenge flow. I'm reverting this change.